oaknational/Oak-Web-Application
# ENG-478 Feat/lesson links scroll

## Description

Started this when making the canonical lesson pages, thought would be nice to get it in before launch.

- page anchor links now are `kebab-case` (generally everything in url should be)
- adds a `useCurrentSection` hook so that the page links follow the scroll position
- clicking page links no longer push an item to the browser history

## How to test

1. Go to https://deploy-preview-1976--oak-web-application.netlify.thenational.academy/teachers/lessons/transverse-waves
2. scroll down
3. You should see the buttons update to signal the are the current section, and the hash update in the url
